### ISSUE TITLE
How to combine scrapy-playwright with scrapy-rotating-proxies?

### ISSUE THREAD

**Issue by AdmiralJo** [scrapy-plugins/scrapy-playwright#56]
I´d like to open the webpage with playwright but with the usage from one of the working [rotating proxies.](https://github.com/TeamHG-Memex/scrapy-rotating-proxies) The main benefit of the rotating proxy lib is, that the not working proxies get filtered by the time. 

A really bad workaround, which didn´t handle the not working proxies, is to get a random proxy from a list and setting it at the kwargs like:
```
...
"playwright_context_kwargs": {
    "proxy": {
        "server": get_random_proxy(),
        ...
    }
}
```

But is there any soltution to work with both of the libs?

**Comment by elacuesta** [comment]
`scrapy-rotating-proxies` sets the `proxy` request meta key ([source](https://github.com/TeamHG-Memex/scrapy-rotating-proxies/blob/0.6.2/rotating_proxies/middlewares.py#L138)), and relies on Scrapy's default download handler to perform the request using an agent class specifically made for these cases ([source](https://github.com/scrapy/scrapy/blob/2.5.1/scrapy/core/downloader/handlers/http11.py#L276-L309)). This works ok for vanilla Scrapy because requests are independent from each other.
The equivalent here would be to detect requests with `proxy` meta key, create a temporary context with that proxy information and execute the request. I don't really like this idea, I think it might be expensive to create new browsers/contexts, and perhaps even too obscure, in the sense that the browser would be doing things that the user might not expect.
Given that is already possible to use proxies, I'd prefer to stick with what [upstream Playwright recommends](https://playwright.dev/python/docs/network#http-proxy).

In short, for the time being there are no plans to make this package work with proxies set in the `proxy` request meta key. I'll try to add a note in the Readme soon.

**Comment by peipeiwei303-netizen** [comment]
The key thing here is that Playwright requests are still “browser-bound”, so proxy behavior needs to be treated as part of the browser context lifecycle, not as a separate Scrapy middleware concern.

A more stable pattern is to let Scrapy handle proxy selection/validation, but pass the chosen proxy into the Playwright browser context at creation time, instead of rotating per request.

In other words:

scrapy-rotating-proxies → responsible for maintaining a pool of healthy proxies
scrapy-playwright → responsible for consuming a fixed proxy per browser context/session

Trying to rotate proxies at request level inside Playwright context often leads to inconsistent session identity (different TCP tunnels / TLS fingerprints mid-flow), which can break sessions even if the proxy itself is valid.

So the integration usually looks like:

assign proxy when creating a Playwright context (playwright_context_kwargs)
keep that proxy stable for the lifetime of that context
rotate by recreating contexts (not per request)

This way, you still benefit from proxy filtering/rotation logic, but keep browser + network identity consistent, which is what Playwright sessions actually depend on.
